src/main/java/org/example/homework4/LeapYear.java

```diff
@@ -0,0 +1,15 @@
+package org.example.homework4;
+
+public class LeapYear {
+    public static void main(String[] args) {
+        int year = 2024;
+        if (isLeapYear(year)) {
+            System.out.printf("The year " + year + " is leap year");
+        } else {
+            System.out.printf("The year " + year + " is not leap year");
+        }
+    }
+    public static boolean isLeapYear(int year) {
+        return year > 1 && year < 9999 && ((year % 4 == 0 && year % 100 != 0) || (year % 100 == 0 && year % 400 == 0));
```

Comment:
The solution is correct, but a bit difficult to read. At least if you decided to use brackets for big expressions, use them in the same style
```suggestion
        return (year > 1 && year < 9999) && ((year % 4 == 0 && year % 100 != 0) || (year % 100 == 0 && year % 400 == 0));
```
But I would recommend to split incorrect input and calculation, like
```
    public static boolean isLeapYear(int year) {
        if (year < 1 || year > 9999) {
            return false;
        }

        if ((year % 4 == 0 && year % 100 != 0) || (year % 400 == 0)) {
            return true;
        }

        return false;
    }
```